examples/step-12/step-12.cc

```diff
@@ -571,34 +532,37 @@ namespace Step12
   }
 
 
-  // The output of this program consists of eps-files of the adaptively refined
-  // grids and the numerical solutions given in gnuplot format.
+  // The output of this program consists of a vtk file of the adaptively
+  // refined grids and the numerical solutions. Finally, we also compute the
+  // L-infinity norm of the solution using VectorTools::integrate_difference.
   template <int dim>
   void AdvectionProblem<dim>::output_results(const unsigned int cycle) const
   {
-    // First write the grid in eps format.
-    {
-      const std::string filename = "grid-" + std::to_string(cycle) + ".eps";
-      deallog << "Writing grid to <" << filename << ">" << std::endl;
-      std::ofstream eps_output(filename);
+    const std::string filename = "solution-" + std::to_string(cycle) + ".vtk";
+    std::cout << "  Writing solution to <" << filename << ">" << std::endl;
+    std::ofstream output(filename);
 
-      GridOut grid_out;
-      grid_out.write_eps(triangulation, eps_output);
-    }
-
-    // Then output the solution in gnuplot format.
-    {
-      const std::string filename = "sol-" + std::to_string(cycle) + ".gnuplot";
-      deallog << "Writing solution to <" << filename << ">" << std::endl;
-      std::ofstream gnuplot_output(filename);
+    DataOut<dim> data_out;
+    data_out.attach_dof_handler(dof_handler);
+    data_out.add_data_vector(solution, "u");
 
-      DataOut<dim> data_out;
-      data_out.attach_dof_handler(dof_handler);
-      data_out.add_data_vector(solution, "u");
+    data_out.build_patches();
 
-      data_out.build_patches();
+    data_out.write_vtk(output);
 
-      data_out.write_gnuplot(gnuplot_output);
+    {
+      Vector<float> values(triangulation.n_active_cells());
+      VectorTools::integrate_difference(dof_handler,
+                                        solution,
+                                        ZeroFunction<dim>(),
```

Comment:
What's the point of this operation? To show stability of the discretization? I think it would be nice to say that in the function documentation.

Reply:
good point, I totally forgot about it.